### DISCUSSION TITLE
Mimic Upgrade

### DISCUSSION THREAD

**Discussion by bkoee** [P3pp3rF1y/SophisticatedBackpacks#1097] [Ideas]
My idea to combat the many requests for mod integrations for upgrades is the mimic upgrade. I'm not sure it's possible but you would craft the mimic upgrade and point it to an item (vanilla or modded). Your mod scans the item for interactions/uses and incorporates it as an upgrade. Example item:ender chest. Mimic upgrade takes the ender chest icon and incorporates it into the upgrade Icon, when upgrade tab is pressed, opens ender chest. Not sure if this is feasible or something you want to explore because it fundamentally changes your mod and I don't want to take away from your hard work and vision

**Comment by P3pp3rF1y** [comment]
this is basically not possible or at least not without implement a complex code which would need to be also modified to support all kinds of items and that would take way more time to do than implementing the individual upgrades

**Reply by bkoee** [reply]
Understood! Thank you for your time :)
